src/modules/rubberband/filter_rbpitch.cpp

```diff
@@ -66,29 +66,62 @@ static int rbpitch_get_audio( mlt_frame frame, void **buffer, mlt_audio_format *
 	int error = mlt_frame_get_audio( frame, buffer, format, frequency, channels, samples );
 	if ( error ) return error;
 
+	// Make sure the audio is in the correct format
+	// This is useful if the filter is encapsulated in a producer and does not
+	// have a normalizing filter before it.
+	if (*format != mlt_audio_float && frame->convert_audio != NULL)
+	{
+		frame->convert_audio( frame, buffer, format, mlt_audio_float );
+	}
+
+	// Detect the last frame
 	bool last_frame = false;
 	if ( mlt_filter_get_position( filter, frame ) + 1 == mlt_filter_get_length2( filter, frame ) )
 	{
 		// This is the last mlt frame. Drain the rest of the stretcher
 		last_frame = true;
 	}
 
+	// Sanity check parameters
```

Comment:
This is my solution for graceful degradation of extreme parameters. These are the result of trial and error. They could be easily tweaked in the future.